tsdb/chunkenc/histo.go

```diff
@@ -258,9 +258,84 @@ func (a *HistoAppender) Appendable(h histogram.SparseHistogram) ([]Interjection,
 	if !ok {
 		return nil, nil, false
 	}
+	if counterReset(a.posbuckets, h.PositiveBuckets, a.posSpans, h.PositiveSpans) {
+		return nil, nil, false
+	}
+	if counterReset(a.negbuckets, h.NegativeBuckets, a.negSpans, h.NegativeSpans) {
+		return nil, nil, false
+	}
+
 	return posInterjections, negInterjections, ok
 }
 
+// counterReset returns true if there was a counter reset for any bucket.
+// This should be called only when buckets are same or new buckets were added,
+// and does not handle the case of buckets missing.
+func counterReset(oldBuckets, newBuckets []int64, oldSpans, newSpans []histogram.Span) bool {
+	if len(oldSpans) == 0 || len(oldBuckets) == 0 {
+		return false
+	}
+
+	oldSpanSliceIdx, newSpanSliceIdx := 0, 0                   // Index for the span slices.
+	oldInsideSpanIdx, newInsideSpanIdx := uint32(0), uint32(0) // Index inside a span.
+	oldIdx := oldSpans[0].Offset
+	newIdx := newSpans[0].Offset
+
+	oldBucketSliceIdx, newBucketSliceIdx := 0, 0 // Index inside bucket slice.
+	oldVal, newVal := oldBuckets[oldBucketSliceIdx], newBuckets[newBucketSliceIdx]
```

Comment:
Isn't it clearer to say `oldVal, newVal := oldBuckets[0], newBuckets[0]` (as above in L281 and 282)?